src/java.base/share/classes/jdk/internal/classfile/impl/StackMapGenerator.java

```diff
@@ -1093,11 +1093,15 @@ void copyFrom(Frame src) {
         }
 
         void checkAssignableTo(Frame target) {
+            int localsSize = this.localsSize;
+            int stackSize = this.stackSize;
             if (target.flags == -1) {
                 target.locals = locals == null ? null : Arrays.copyOf(locals, localsSize);
                 target.localsSize = localsSize;
-                target.stack = stack == null ? null : Arrays.copyOf(stack, stackSize);
-                target.stackSize = stackSize;
+                if (stackSize > 0) {
+                    target.stack = Arrays.copyOf(stack, stackSize);
```

Comment:
Do you think we should use `.clone()` to avoid `getClass` checks and to avoid extra array growing in case the new stack/local immediately grows due to instructions after target?

Also I think you can do `localsSize > 0` check for locals assignment too.

Reply:
stack.length may be larger than stackSize. In this case, the length of the array copied by clone will be longer than Arrays.copyOf. However, I have no evidence that Arrays.copyOf is faster than clone.

Reply:
I found through debugging that in the `target.flags == -1` branch of checkAssignableTo, `localsSize > 0 and stackSize = 0` are always, so I made this optimization.

Reply:
`stackSize` always has to be less than or equal to `stack.length`; this is just like `ArrayList`, where `stackSize` is the size and `stack.length` is capacity.